apps/communications/contacts/js/views/multiple_select.js

```diff
@@ -0,0 +1,136 @@
+
+Contacts.MultipleSelect = (function() {
+  const STATUS_TIMEOUT = 3000; // ms before closing view which shows status msg
+  var contactsToImport = [];
+  var header, saveButton, container, template, title;
+
+  function init() {
+    header = document.getElementById('multiple-select-view-header');
+    saveButton = document.getElementById('save-button');
+    container = document.getElementById('multiple-select-container');
+    template = document.getElementById('contactToImportTemplate');
+    title = document.getElementById('multiple-select-view-title');
+
+    saveButton.addEventListener('click', importAll);
+    header.addEventListener('action', cancelHandler);
+  }
+
+  function cancelHandler() {
+    header.removeEventListener('action', cancelHandler);
+    saveButton.removeEventListener('click', importAll);
+    Contacts.cancel();
+  }
+
+  function importAll() {
+    var numDupsMerged = 0,
+        importedContacts = 0;
+    const DEPENDENCIES = [
+      '/shared/js/contacts/import/utilities/status.js',
+      '/shared/js/simple_phone_matcher.js',
+      '/shared/js/contacts/contacts_matcher.js',
+      '/shared/js/contacts/contacts_merger.js',
+      '/shared/js/contacts/merger_adapter.js'
+    ];
+    LazyLoader.load(DEPENDENCIES, () => {
+      contactsToImport.forEach((contact, index) => {
+        contacts.Matcher.match(contact, 'passive', {
+          onmatch: (matches) => {
+            var callbacks = {
+              success: () => {
+                numDupsMerged++;
+                doContinue(index, true);
+              },
+              error: () => doContinue(index)
+            };
+            contacts.adaptAndMerge(contact, matches, callbacks);
+          },
+          onmismatch: () => {
+            var saving = navigator.mozContacts.save(contact);
+
+            saving.onsuccess = () => {
+              doContinue(index, true);
+            };
+
+            saving.onerror = (err) => {
+              console.error(err);
+              doContinue(index);
+            };
+          }
+        });
+      });
+    });
+
+
+    function doContinue(index, isContactImported) {
+      if (isContactImported) {
+        importedContacts++;
+      }
+
+      if (contactsToImport.length - 1 <= index) {
+        utils.status.show({
+          id: 'vCardContacts-imported',
+          args: {
+            n: importedContacts
+          }
+        }, {
+          id: 'contactsMerged',
+          args: {
+            numDups: numDupsMerged
+          }
+        });
+
+        window.setTimeout(cancelHandler, STATUS_TIMEOUT);
+      }
+    }
+  }
+
+  /**
+   * Allows to preview which contacts will be imported.
+   *
+   * @param {mozContact[]} contacts - Every contact already read from vCard.
+   * @param cursor - Reader object to continue reading from vCard.
+   * @param {string} filename - Name of the vcf file we are reading from.
+   */
+  var render = function(contacts, cursor, filename) {
+    init();
```

Comment:
My idea is to left it there and not expose it to the outside so init is executed when the DOM is ready, which is problematic in tests.